Scripts/script-GetDuplicatesMl.yml

```diff
@@ -0,0 +1,641 @@
+          return self.memo[args]
+
+
+  class Utils():
+      email_pattern = re.compile(r"([a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+)")
+
+      @staticmethod
+      def extract_domain_from_url(url):
+          parsed_domain = urlparse(url)
+          domain = parsed_domain.netloc
+          domain_parts = domain.split('.')
+          if len(domain_parts) > 2:
+              domain = '.'.join(domain_parts[-(2 if domain_parts[-1] in {
+                  'com', 'net', 'org', 'io', 'ly', 'me', 'sh', 'fm', 'us'} else 3):])
+          if len(domain) == 0:
+              return None
+          return domain
+
+      @staticmethod
+      def uri_validator(url):
+          try:
+              result = urlparse(url)
+              if result.scheme and result.netloc and result.path:
+                  return True
+          except:
+              return False
+
+      @staticmethod
+      def extract_email_domain(email):
+          index = email.rfind("@")
+          if index == -1:
+              return None
+          return email[index + 1:]
+
+      @staticmethod
+      def extract_email_username(email):
+          index = email.rfind("@")
+          if index == -1:
+              return None
+          return email[:index]
+
+      @staticmethod
+      def get_url_path(url):
+          return urlparse(url).path
+
+      @staticmethod
+      def ip_address_in_network(ip, net):
+          try:
+              ipaddr = int(''.join(['%02x' % int(x) for x in ip.split('.')]), 16)
+              netstr, bits = net.split('/')
+              netaddr = int(''.join(['%02x' % int(x) for x in netstr.split('.')]), 16)
+              mask = (0xffffffff << (32 - int(bits))) & 0xffffffff
+              return (ipaddr & mask) == (netaddr & mask)
+          except Exception:
+              return None
+
+      @staticmethod
+      def get_time_diff_seconds(date1, date2):
+          try:
+              if 'datetime' in str(type(date1)):
+                  return abs(date1 - date2).total_seconds()
+              else:
+                  return abs(dateutil.parser.parse(date1) - dateutil.parser.parse(date2)).total_seconds()
+          except Exception:
+              return None
+
+      @staticmethod
+      def get_incident_labels_map(labels, labels_rename=EMAIL_LABELS_MAP):
+          if labels is None:
+              return {}
+          labels_map = {}
+          for label in labels:
+              label_type = label['type']
+              if label_type in labels_rename:
+                  label_type = labels_rename[label_type]
+              labels_map[label_type] = label['value']
+          return labels_map
+
+      @staticmethod
+      def get_domains(indicators, labels):
+          domains = []
+          for email in indicators.get('Email', []):
+              domains.append(Utils.extract_email_domain(email))
+
+          for url in indicators.get('URL', []):
+              domains.append(Utils.extract_domain_from_url(url))
```

Comment:
there might be indicator of type 'Domain', you need to user that too, generated by:
https://github.com/demisto/content/blob/master/Scripts/script-ExtractDomain.yml

Reply:
handle this

Reply:
V